File: LC/Binary_Trees/94.java
package LC.Binary_Trees;

public class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        if (root == null){
            return new ArrayList<>();
        }
        List<Integer> result = new ArrayList<>();
        result.addAll(inorderTraversal(root.left));
        result.add(root.val);
        result.addAll(inorderTraversal(root.right));
        return result;
    }
}

//Binary Tree Inorder Traversal

File: LC/Binary_Trees/94.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        if (root == nullptr) return {};
        vector<int> left = inorderTraversal(root->left);
        vector<int> right = inorderTraversal(root->right);
        vector<int> result;
        result.insert(result.end(),left.begin(),left.end());
        result.push_back(root->val);
        result.insert(result.end(),right.begin(),right.end());
        return result;
    }
};

//Binary Tree Inorder Traversal
